### DISCUSSION TITLE
Collaborating with a Brazilian Portuguese version.

### DISCUSSION THREAD

**Discussion by lucaspicinini** [ishtms/learn-nodejs-hard-way#31] [Ideas]
Hey bro, first of all thanks for your work, it has helped me a lot with a college subject. I'm Brazilian and I ended up interested in translating your book. I believe that there is a lack of content of this level in PT-BR and that we can make the material accessible to many Brazilians and other countries that speak this language (Portugal, Angola, Cape Verde, Guinea Bissau, São Tomé)... Check the pt-br branch of my fork to see the beginning of the translation. It will take me a few weeks to complete, because due to several factors, I can only dedicate around 1 hour to translating. I don't want credit for anything other than the translation, I wanted to make it clear to people who come to my repository that it is just a translation of your initiative. What would be the best way to do this? Could you help me with this organization? I've only been programming for 7 months and I'm in the first period of college, just a newbie lol, but I do everything with a lot of love, dedication and I already have experience with translations due to a channel I have on YouTube (@brazilianmmalegends). Hugs from BR!

**Comment by ishtms** [comment]
Hi @lucaspicinini,

Thank you your message. It's great to hear that this repo/guide has been helpful with your college work.

Also, it's really awesome that you're interested in translating the content into Portuguese. It will be awesome to make the material more accessible to more people.

If you want to make your branch visible to others on this repository, you'll need to submit a pull request. I've already taken a look at the branch on your repo.

**Reply by lucaspicinini** [reply]
Thanks for the response @ishtms , I'm glad you liked the idea. I would definitely like it, the intention is to make it visible to as many people as possible. I have some doubts. It would be better to make a pull request only after the translation is finished, right?! Would this pull request be for the master branch of your repository or would you create another one for that? I'm sorry for asking so many basic questions, but I'm still a white belt at this lol. Until next time, have a great day!!!

**Reply by ishtms** [reply]
@lucaspicinini I've created a new branch `pt-br`, please open a PR, I'll accept it. One more thing I'd like to add is, keep one file per commit. In short, please do not include more than one file per commit. That way it would be easier for me to test things and make sure they're OK. 

Once again, thanks again for your work! Really appreciate it.

**Reply by lucaspicinini** [reply]
Okay @ishtms , I'll keep things the way you said. Due to my little experience, two of the commits (`b40b625` and `53e21f0`) were made with more than 1 file, but I believe you will be able to understand, as I will make some clarifications below to make it easier. I made the pull request with what I already have, but from now on, I will keep one file per commit as you indicated and try to commit only with the completed file; and only then make a new pull request. I believe this way it will be more organized for you. If you have any other preferences or tips for the workflow, just let me know here.

Now, about the translation. Briefly:

./Readme.md (100% translated)
./chapters/Readme.md (100%)
./chapters/ch00 (around 35%)
./chapters/ch01 (100%)
./chapters/ch02 (100%)

So far, in this first pull request, these were the files changed. I haven't changed any file names too, I thought it would be better to leave this work to be done at the end, that way the link structure wouldn't be affected during translation. Only raw text was translated and in the ch02 file, I translated the string "Let's learn Javascript" that was in the code example into Portuguese, but the logic of the code, the meaning of the example or your way of teaching was not affected.

I hope things are clear, if not, let me know here. Hugs from Brazil and have a great day.

**Comment by lucaspicinini** [comment]
Hey @ishtms , I hope this start of year has been incredible for you and I wish you much success throughout 2024. I received notifications of the last merge into the master branch and I would like to inform you that I will try to keep this translation updated for as long as I can, because it was a book that helped me a lot. I'm not sure if I wait for the first PR to complete or if it would be better to add the new information, like your channel, etc... in new commits in that same pull request. Feedback on how to maintain new updates in the main branch would help me with this question. Have a great week!!!
